janus_core/geom_opt.py

```diff
@@ -60,35 +69,56 @@ def optimize(
     dyn_kwargs = dyn_kwargs if dyn_kwargs else {}
     filter_kwargs = filter_kwargs if filter_kwargs else {}
     opt_kwargs = opt_kwargs if opt_kwargs else {}
-    struct_kwargs = struct_kwargs if struct_kwargs else {}
+    write_kwargs = write_kwargs if write_kwargs else {}
     traj_kwargs = traj_kwargs if traj_kwargs else {}
+    log_kwargs = log_kwargs if log_kwargs else {}
 
-    if struct_kwargs and "filename" not in struct_kwargs:
-        raise ValueError("'filename' must be included in struct_kwargs")
+    if write_kwargs and "filename" not in write_kwargs:
+        raise ValueError("'filename' must be included in `write_kwargs`")
 
     if traj_kwargs and "filename" not in traj_kwargs:
-        raise ValueError("'filename' must be included in traj_kwargs")
+        raise ValueError("'filename' must be included in `traj_kwargs`")
 
     if traj_kwargs and "trajectory" not in opt_kwargs:
         raise ValueError(
-            "'trajectory' must be a key in opt_kwargs to save the trajectory."
+            "'trajectory' must be a key in `opt_kwargs` to save the trajectory."
         )
 
+    if log_kwargs and "filename" not in log_kwargs:
+        raise ValueError("'filename' must be included in `log_kwargs`")
+
+    log_kwargs.setdefault("name", __name__)
+    logger = config_logger(**log_kwargs)
+
     if filter_func is not None:
         filtered_atoms = filter_func(atoms, **filter_kwargs)
         dyn = optimizer(filtered_atoms, **opt_kwargs)
+        if logger:
+            logger.info("Using filter %s", filter_func.__name__)
+            logger.info("Using optimizer %s", optimizer.__name__)
+
     else:
         dyn = optimizer(atoms, **opt_kwargs)
 
+    if logger:
+        logger.info("Starting geometry optimization")
+
     dyn.run(fmax=fmax, **dyn_kwargs)
 
     # Write out optimized structure
-    if struct_kwargs:
-        write(images=atoms, **struct_kwargs)
+    if write_results:
+        if "filename" not in write_kwargs:
+            write_kwargs["filename"] = Path(
+                f"./{atoms.get_chemical_formula()}-opt.xyz"
+            ).absolute()
```

Comment:
```suggestion
        write_kwargs.setdefault("filename",
                                Path(f"./{atoms.get_chemical_formula()}-opt.xyz").absolute()
                                )
```
Though given the check above is this possible to reach?